subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -480,7 +481,7 @@ void ull_filter_adv_update(u8_t adv_fp)
 	filter_clear(&rl_filter);
 
 	if (rl_enable &&
-	    IS_ENABLED(CONFIG_BT_OBSERVER) && !ull_scan_is_enabled(0)) {
+	    (!IS_ENABLED(CONFIG_BT_OBSERVER) || !ull_scan_is_enabled(0))) {
```

Comment:
Personal opinion: I think !(IS_ENABLED(CONFIG_BT_OBSERVER) && ull_scan_is_enabled(0)) is easier to read and would have preferred it here.